src/Engine/ProtoCore/Lang/CallSite.cs

```diff
@@ -1912,16 +1910,18 @@ private FunctionGroup FirstFunctionGroupInInheritanceChain(RuntimeCore runtimeCo
             if (ret.IsNull)
             {
                 //wipe the trace cache
-                TraceUtils.ClearTLSKey(TRACE_KEY);
+                TraceUtils.ClearAllKnownTLSKeys();
+                newTraceData.Data = null;
```

Comment:
@saintentropy is checking to see if this is required. I pointed this out to him.